src/org/zaproxy/zap/extension/ruleconfig/RuleConfigParam.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2016 The ZAP Development team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.ruleconfig;
+
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.parosproxy.paros.common.AbstractParam;
+
+public class RuleConfigParam extends AbstractParam {
+
+    private static final String RULES_BASE_KEY = "rules";
+    private static final String RULES_DEFAULT_KEY_EXT = ".default";
+    
+	private Map<String, RuleConfig> ruleConfigs = new HashMap<String, RuleConfig>();
+
+    public RuleConfigParam() {
+    	// Add the built in rule configs
+        this.addRuleConfig(new RuleConfig("rules.common.sleep", "5"));
+    }
+
+    @Override
+    protected void parse() {
+    	
+    	Iterator<String> iter = this.getConfig().getKeys(RULES_BASE_KEY);
+    	while (iter.hasNext()) {
+    		String key = iter.next();
+    		if (ruleConfigs.containsKey(key)) {
+    			ruleConfigs.get(key).setValue(this.getConfig().getString(key));
+    		} else {
+    			// Rules can specify an optional default value in the config file
+    	        this.addRuleConfig(
+    	        		new RuleConfig(
+    						key, 
+    						this.getConfig().getString(key + RULES_DEFAULT_KEY_EXT, ""),
+    						this.getConfig().getString(key)));
+    		}
+    	}
+    }
+
+	public void addRuleConfig (RuleConfig rc) {
+		this.ruleConfigs.put(rc.getKey(), rc);

```

Comment:
Should add the new rule to the config file? (Parse could put directly to `ruleConfigs` in that case,)